url20b0f880a5_jgraph_legacy_jgraph5_tgz-pJ8-org_jgraph_example_HelloWorldJ8/src/org/jgraph/util/ParallelEdgeRouter.java

```diff
@@ -0,0 +1,456 @@
+/*
+ * $Id: ParallelEdgeRouter.java,v 1.4 2008-11-13 00:12:39 david Exp $
+ * 
+ * Copyright (c) 2001-2007 Gaudenz Alder
+ * Copyright (c) 2004-2007 David Benson
+ *  
+ */
+package org.jgraph.util;
+
+import java.awt.geom.Point2D;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.jgraph.JGraph;
+import org.jgraph.graph.AbstractCellView;
+import org.jgraph.graph.CellView;
+import org.jgraph.graph.DefaultGraphModel;
+import org.jgraph.graph.Edge;
+import org.jgraph.graph.EdgeView;
+import org.jgraph.graph.GraphConstants;
+import org.jgraph.graph.GraphLayoutCache;
+import org.jgraph.graph.GraphModel;
+import org.jgraph.graph.DefaultEdge.LoopRouting;
+
+/**
+ * A routing algorithm that 
+ */
+public class ParallelEdgeRouter extends LoopRouting {
+	/**
+	 * Singleton to reach parallel edge router
+	 */
+	protected static final ParallelEdgeRouter sharedInstance = new ParallelEdgeRouter();
+
+	/**
+	 * Distance between each parallel edge
+	 */
+	private static double edgeSeparation = 10.;
+
+	/**
+	 * Distance between intermediate and source/target points
+	 */
+	private static double edgeDeparture = 10.;
+
+	/**
+	 * Getter for singleton managing parallel edges
+	 * 
+	 * @return ParallelEdgeRouter for parallel edges
+	 */
+	public static ParallelEdgeRouter getSharedInstance() {
+		return ParallelEdgeRouter.sharedInstance;
+	}
+
+	/**
+	 * Calc of intermediates points
+	 * 
+	 * @param edge
+	 *            Edge for which routing is demanding
+	 */
+	public List routeEdge(GraphLayoutCache cache, EdgeView edge) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 33. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.